homeassistant/components/homekit_controller/__init__.py

```diff
@@ -304,12 +304,13 @@ def _setup_characteristic(self, char):
     @property
     def unique_id(self):
         """Return the ID of this device."""
-        return self._address
+        serial = self._accessory_info.get('serial-number', 'unknown')
+        return "homekit-{}-{}".format(serial, self._iid)
 
     @property
     def name(self):
         """Return the name of the device if any."""
-        return self._name
+        return self._accessory_info.get('name', None)
```

Comment:
`None` is the default value returned if the key is missing in the dict.

Reply:
Gah, have to stop doing that. (I like to remind myself that None is a thing that might happen)